pkg/client/cache/deploymentconfig.go

```diff
@@ -0,0 +1,44 @@
+package cache
+
+import (
+	"fmt"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/client/cache"
+
+	deployapi "github.com/openshift/origin/pkg/deploy/api"
+	deployutil "github.com/openshift/origin/pkg/deploy/util"
+)
+
+// StoreToDeploymentConfigLister gives a store List and Exists methods. The store must contain only deploymentconfigs.
+type StoreToDeploymentConfigLister struct {
```

Comment:
I think you want a pattern like these: https://github.com/kubernetes/kubernetes/blob/master/pkg/client/cache/listers.go#L43 . I suspect you can do most of it generically and cast at the end.